src/NewTools-Debugger-Tests/StDebuggerTest.class.st

```diff
@@ -506,6 +506,31 @@ StDebuggerTest >> testDebuggerApplication [
 		identicalTo: StDebugger currentApplication
 ]
 
+{ #category : 'tests - session' }
+StDebuggerTest >> testDebuggerDoesNotLogStackToFileSetting [
+
+	| logFileRef debuggingStackToFile |
+	
+	logFileRef := Smalltalk logFileName asFileReference.
+	logFileRef ensureDelete.
+	self deny: logFileRef exists.
+	debuggingStackToFile := session class logDebuggerStackToFile.
+
+	debugger := StTestDebuggerProvider new debuggerWithDNUContext. 
+	session := debugger session.
+	session class logDebuggerStackToFile: false.
+	debugger
+		application: debugger class currentApplication;
+		initialize;
+		open.
+
+	self 
+		assert: logFileRef exists not
+		description: 'It tests that when the log debugger stack to file setting is false, a Pharo log file is not created when requesting a debugger'.
+
+	session class logDebuggerStackToFile: debuggingStackToFile
```

Comment:
If the test fails, it will not set back the `debuggingStackToFile` setting. Maybe, the assignment should be done in an `ensure:` block or maybe it should be set in the `tearDown:`